src/main/java/graphql/scalar/GraphqlFloatCoercing.java

```diff
@@ -29,7 +29,15 @@
 public class GraphqlFloatCoercing implements Coercing<Double, Double> {
 
     private Double convertImpl(Object input) {
-        if (isNumberIsh(input)) {
+        // From the GraphQL Float spec, non-finite floating-point internal values (NaN and Infinity)
+        // must raise a field error on both result and input coercion
+        if (input instanceof Double) {
+            Double doubleInput = (Double) input;
+            if (Double.isNaN(doubleInput) || Double.isInfinite(doubleInput)) {
+                return null;
+            }
+            return doubleInput;
+        } else if (isNumberIsh(input)) {
```

Comment:
I did not add a separate `else if (input instanceof Float)` branch because it causes some tests to fail.

Getting the `doubleValue()` from a BigDecimal produces an ever-so-slightly-different result to getting the `doubleValue()` out of a Float.

Reply:
Current this is

```
private Double convertImpl(Object input) {
        // From the GraphQL Float spec, non-finite floating-point internal values (NaN and Infinity)
        // must raise a field error on both result and input coercion
        if (input instanceof Double) {
            Double doubleInput = (Double) input;
            if (Double.isNaN(doubleInput) || Double.isInfinite(doubleInput)) {
                return null;
            }
            return doubleInput;
        } else if (isNumberIsh(input)) {
            BigDecimal value;
            try {
                value = new BigDecimal(input.toString());
            } catch (NumberFormatException e) {
                return null;
            }
            return value.doubleValue();
        } else {
            return null;
        }
    }
```

So it does not apply the checks to the BigDecimal derived string

Why not do

```
private Double convertImpl(Object input) {
        // From the GraphQL Float spec, non-finite floating-point internal values (NaN and Infinity)
        // must raise a field error on both result and input coercion
        Double doubleInput;
        if (input instanceof Double) {
            doubleInput = (Double) input;
        } else if (isNumberIsh(input)) {
            BigDecimal value;
            try {
                value = new BigDecimal(input.toString());
            } catch (NumberFormatException e) {
                return null;
            }
            doubleInput =  value.doubleValue();
        } else {
            return null;
        }
        
        if (Double.isNaN(doubleInput) || Double.isInfinite(doubleInput)) {
                return null;
        }
       return doubleInput
    }
```

Forgive me if this doesnt compile

Reply:
This seems more sensible - eg first get it into a double and then check it - regardless

Even if BigDecimal never produces Nan or infinity, the logic reads better

Reply:
Womp womp

So it can produce infinity! This was a great idea